python/taichi/lang/kernel_impl.py

```diff
@@ -368,6 +369,8 @@ def extract_arguments(self):
                     pass
                 elif id(annotation) in primitive_types.type_ids:
                     pass
+                elif id(annotation) == id(SparseMatrixBuilder):
```

Comment:
```suggestion
                elif isinstance(annotation, SparseMatrixBuilder):
```

Reply:
When I use `isinstance`,  annotation errors occur. It woundn't happen if using `id`.

Reply:
Since we are using `ti.SparseMatrix` (instead of `ti.SparseMatrix()`) here as the annotation, the annotation is the class itself instead of an instance. 

Not sure if this is the right design decision though.